housekeeper/store/database.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, Session, sessionmaker
+
+from housekeeper.exc import HousekeeperError
+from housekeeper.store.models import Model
+
+
+SESSION_FACTORY: Optional[Session] = None
+ENGINE = None
+
+
+def initialise_database(db_uri: str):
+    """Initialize the global engine and session factory for SQLAlchemy."""
+    global SESSION_FACTORY, ENGINE
+    ENGINE = create_engine(db_uri, pool_pre_ping=True)
```

Comment:
type hints

Reply:
Typed at the top of the file now where it is "declared"